providers/shared/components/directory/id.ftl

```diff
@@ -0,0 +1,106 @@
+[#ftl]
+
+[@addComponentDeployment
+    type=DIRECTORY_COMPONENT_TYPE
+    defaultGroup="solution"
+/]
+
+[@addComponent
+    type=DIRECTORY_COMPONENT_TYPE
+    properties=
+        [
+            {
+                "Type"  : "Description",
+                "Value" : "A managed directory services instance"
+            }
+        ]
+    attributes=
+        [
+            {
+                "Names" : "Engine",
+                "Types" : STRING_TYPE,
+                "Values" : [ "Simple", "ActiveDirectory" ],
+                "Mandatory" : true
+            },
+            {
+                "Names" : [ "Certificate", "Hostname" ],
+                "Children" : certificateChildConfiguration
+            },
+            {
+                "Names" : "IPAddressGroups",
+                "Types" : ARRAY_OF_STRING_TYPE,
+                "Default" : []
+            },
+            {
+                "Names" : "ShortName",
+                "Types" : STRING_TYPE
+            },
+            {
+                "Names" : "Size",
+                "Types" : STRING_TYPE
+            },
```

Comment:
Is this required? Or does it have specific value?

Reply:
It has different values depending on whether you are looking at SimpleAD (Small or Large), or MicrosoftAD (Standard or Enterprise). Was that not what you were after?

Yes it probably should be mandatory. Done

Reply:
As a general user does that value matter? For both services aren't they just the physical size of the deployment? Almost like the Processor profile size? 

So could we just go small, medium, large, and then in the backend handle translating those to the appropriate service?

Reply:
For MicrosoftAD, it's also whether it's cross Region, but mostly sizing. Each only has two options - so small and large?

Reply:
Yeah I think lets go Small/Large and we can worry about multi region separately 
That would be an AWS specific thing and we could make that a check between the values

Reply:
Done